File: app/src/main/java/com/fieldbook/tracker/brapi/BrAPIServiceFactory.java
package com.fieldbook.tracker.brapi;

import android.content.Context;

import com.fieldbook.tracker.database.DataHelper;
import com.fieldbook.tracker.preferences.GeneralKeys;
import com.fieldbook.tracker.utilities.Constants;

public class BrAPIServiceFactory {

    public static final BrAPIService getBrAPIService(Context context){

        String version = context.getSharedPreferences("Settings", 0)
                .getString(GeneralKeys.BRAPI_VERSION, "V1");
        BrAPIService brAPIService;
        if(version.equals("V2"))
            brAPIService = new BrAPIServiceV2(context);
        else
            brAPIService = new BrAPIServiceV1(context);

        return brAPIService;
    }
}

File: shared/src/commonMain/kotlin/com/fieldbook/shared/brapi/BrAPIServiceFactory.kt
package com.fieldbook.shared.brapi

object BrAPIServiceFactory {
    const val VERSION_V1 = "V1"
    const val VERSION_V2 = "V2"

    fun create(
        baseUrl: String,
        bearerToken: String? = null,
        version: String? = VERSION_V2,
    ): BrAPIService {
        return when (version?.trim()?.uppercase()) {
            VERSION_V1 -> BrAPIServiceV1(
                baseUrl = baseUrl,
                bearerToken = bearerToken,
            )
            else -> BrAPIServiceV2(
                baseUrl = baseUrl,
                bearerToken = bearerToken,
            )
        }
    }
}
